src/Hpack/Config.hs

```diff
@@ -996,8 +996,8 @@ traverseSectionAndConditionals fData fConditionals acc0 sect@Section{..} = do
     traverseConditionals = traverse . traverse . traverseSectionAndConditionals fConditionals fConditionals
 
 getMentionedLibraryModules :: LibrarySection -> [String]
-getMentionedLibraryModules LibrarySection{..} =
-  fromMaybeList librarySectionExposedModules ++ fromMaybeList librarySectionOtherModules
+getMentionedLibraryModules (LibrarySection _ exposedModules generatedExposedModules otherModules generatedOtherModules _ _)
+  = fromMaybeList (exposedModules <> generatedExposedModules <> otherModules <> generatedOtherModules)
```

Comment:
Generated modules are never inferred because they are explicitly mentioned.  This fact was duplicated in several places before.